tgui/packages/tgui/interfaces/PreferencesMenu/loadout/index.tsx

```diff
@@ -0,0 +1,337 @@
+                }
+                onClick={() => {
+                  setSelectedTab(curTab.name);
+                  setSearchLoadout('');
+                }}
+              >
+                <Box>
+                  {curTab.category_icon && (
+                    <Icon name={curTab.category_icon} mr={1} />
+                  )}
+                  {curTab.name}
+                </Box>
+              </Tabs.Tab>
+            ))}
+          </Tabs>
+        </Section>
+      </Stack.Item>
+      <Stack.Item>
+        <LoadoutTabs
+          loadout_tabs={loadout_tabs}
+          currentTab={selectedTabName}
+          currentSearch={searchLoadout}
+          modifyItemDimmer={modifyItemDimmer}
+          setModifyItemDimmer={setModifyItemDimmer}
+        />
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const LoadoutTabs = (props: {
+  loadout_tabs: LoadoutCategory[];
+  currentTab: string;
+  currentSearch: string;
+  modifyItemDimmer: LoadoutItem | null;
+  setModifyItemDimmer: (dimmer: LoadoutItem | null) => void;
+}) => {
+  const {
+    loadout_tabs,
+    currentTab,
+    currentSearch,
+    modifyItemDimmer,
+    setModifyItemDimmer,
+  } = props;
+  const activeCategory = loadout_tabs.find((curTab) => {
+    return curTab.name === currentTab;
+  });
+  const searching = currentSearch.length > 1;
+
+  return (
+    <Stack fill height="550px">
+      <Stack.Item align="center" width="250px" height="100%">
+        <Stack vertical height="100%">
+          <Stack.Item height="60%">
+            <LoadoutPreviewSection />
+          </Stack.Item>
+          <Stack.Item grow>
+            <LoadoutSelectedSection
+              all_tabs={loadout_tabs}
+              modifyItemDimmer={modifyItemDimmer}
+              setModifyItemDimmer={setModifyItemDimmer}
+            />
+          </Stack.Item>
+        </Stack>
+      </Stack.Item>
+      <Stack.Item grow>
+        {searching || activeCategory?.contents ? (
+          <Section
+            title={searching ? 'Searching...' : 'Catalog'}
+            fill
+            scrollable
+            buttons={
+              activeCategory?.category_info ? (
+                <Box italic mt={0.5}>
+                  {activeCategory.category_info}
+                </Box>
+              ) : null
+            }
+          >
+            <Stack vertical>
+              <Stack.Item>
+                {searching ? (
+                  <SearchDisplay
+                    loadout_tabs={loadout_tabs}
+                    currentSearch={currentSearch}
+                  />
+                ) : (
+                  <LoadoutTabDisplay category={activeCategory} />
+                )}
+              </Stack.Item>
+            </Stack>
+          </Section>
+        ) : (
+          <Section fill>
+            <Box>No contents for selected tab.</Box>
+          </Section>
+        )}
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const TypepathToLoadoutItem = (
```

Comment:
```suggestion
const typepathToLoadoutItem = (
```
only components that return jsx get pascal case

as for your comment below it's probably faster with object but not enough of an issue to be necessary. the bundler will also mangle this btw. usememo is another route too if you were worried about it

https://react.dev/reference/react/useMemo

Reply:
yep